gdx/src/com/badlogic/gdx/math/collision/Ray.java

```diff
@@ -34,10 +35,15 @@ public Ray () { }
 	/** Constructor, sets the starting position of the ray and the direction.
 	 * 
 	 * @param origin The starting position
-	 * @param direction The direction */
+	 * @param direction The direction(must be normalized before) */
 	public Ray (Vector3 origin, Vector3 direction) {
 		this.origin.set(origin);
-		this.direction.set(direction).nor();
+		this.direction.set(direction);
+	}
+
+	/** Constructor, sets the starting position of the ray and the direction. */
+	public Ray (Vector3 origin, Vector3 direction, boolean normalizeDirection) {
+		this(origin, normalizeDirection ? direction.nor() : direction);

```

Comment:
This will change the `direction` vector that was passed in, please don't do that. Simple
```java
this(origin, direction);
if (normalizeDirection) {
    this.direction.nor();
}
```
should suffice. Same for `setNormalize`.

Reply:
done !